### PR TITLE
fix: do not duplicate documentation pages on update import

### PR DESCRIPTION
**Pull request by a-cordier**
see https://github.com/gravitee-io/issues/issues/6694

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-kpdcxrargj.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6694-fix-page-duplicates/index.html)
_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PageServiceImpl.java [hunk 1]
```diff
@@ -2659,6 +2657,7 @@ private void createOrUpdateChildrenPages(
             }
 
             if (createdOrUpdatedPage == null) {
+                String newPageEntityId = UuidString.generateForEnvironment(environmentId, apiId, pageEntityToImport.getId());
```
Comment (marcambier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PageServiceImpl.java]: Do we really have to regenerate the page ID here ?

I believe that, if necessary to regenerate ID, it has been done once in the `preprocessApiDefinitionUpdatingIds``;
And so, here, the id in pageEntityToImport is already the one that has been regenerated.

So maybe it's unecessary to regenerate a `newPageEntityId` again here ; and use pageEntityToImport.getId() instead ?

**Comment by gaetanmaisse** [comment]
To discuss, I did some tests and pages inside a subfolder were lost after an API promotion

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java [hunk 2]
```diff
@@ -107,12 +107,16 @@ public ApiEntity createWithImportedDefinition(String apiDefinitionOrURL, String
         String apiDefinition = fetchApiDefinitionContentFromURL(apiDefinitionOrURL);
         try {
             final JsonNode jsonNode = objectMapper.readTree(apiDefinition);
+
+            String apiId = UuidString.generateRandom();
+            ((ObjectNode) jsonNode).put("id", apiId);
```
Comment (marcambier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java]: I wonder if that break the behavior of the line 3 of our "Import rules" GoogleSheets file : `200 : API is created with provided ID`

Comment (a-cordier) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java]: That was messing with the test suite indeed. https://github.com/gravitee-io/gravitee-api-management/pull/1005/commits/bd61499eda4c7492d00fd3a04699225b476448ca makes tests pass.

File: gravitee-apim-cypress/cypress/integration/rest-api/apis/import-update-tests.ts [hunk 3]
```diff
@@ -460,6 +460,43 @@ context('API - Imports - Update', () => {
         deleteApi(ADMIN_USER, apiId).noContent();
       });
     });
+
+    describe.only('Update API, duplicating system folder', () => {
```
Comment (marcambier) [inline gravitee-apim-cypress/cypress/integration/rest-api/apis/import-update-tests.ts]:
```suggestion
    describe('Update API, duplicating system folder', () => {
```

**Review by marcambier — CHANGES_REQUESTED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java (hunk 2 above)
Comment (marcambier) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java]: ok

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PageServiceImpl.java (hunk 1 above)
Comment (marcambier) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PageServiceImpl.java]: ok

**Review by marcambier — APPROVED** [review]

**Pull request merged**